From fb2c8a946384ac07ad74ec3df7ee5906ffbf0b72 Mon Sep 17 00:00:00 2001
From: Markus Armbruster <armbru@redhat.com>
Date: Sat, 2 Nov 2013 10:01:27 +0100
Subject: [PATCH 27/29] smbios: Decouple system product from QEMUMachine

RH-Author: Markus Armbruster <armbru@redhat.com>
Message-id: <1383386488-29789-11-git-send-email-armbru@redhat.com>
Patchwork-id: 55247
O-Subject: [PATCH 7.0 qemu-kvm 10/11] smbios: Decouple system product from QEMUMachine
Bugzilla: 994490
RH-Acked-by: Michael S. Tsirkin <mst@redhat.com>
RH-Acked-by: Laszlo Ersek <lersek@redhat.com>
RH-Acked-by: Miroslav Rezanina <mrezanin@redhat.com>

From: Markus Armbruster <armbru@redhat.com>

Michael Tsirkin doesn't trust us to keep values of QEMUMachine member
product stable in the future.  Use copies instead, and in a way that
makes it obvious that they're guest ABI.

Note that we can be trusted to keep values of member name, because
that has always been ABI.

Signed-off-by: Markus Armbruster <armbru@redhat.com>
(cherry picked from pending upstream submission)
---
 hw/i386/pc_piix.c | 3 ++-
 hw/i386/pc_q35.c  | 3 ++-
 2 files changed, 4 insertions(+), 2 deletions(-)

Signed-off-by: Miroslav Rezanina <mrezanin@redhat.com>
---
 hw/i386/pc_piix.c |    3 ++-
 hw/i386/pc_q35.c  |    3 ++-
 2 files changed, 4 insertions(+), 2 deletions(-)

diff --git a/hw/i386/pc_piix.c b/hw/i386/pc_piix.c
index e8fe607..396f4e4 100644
--- a/hw/i386/pc_piix.c
+++ b/hw/i386/pc_piix.c
@@ -116,7 +116,8 @@ static void pc_init1(QEMUMachineInitArgs *args,
     }
 
     if (smbios_type1_defaults) {
-        smbios_set_type1_defaults("QEMU", args->machine->desc,
+        /* These values are guest ABI, do not change */
+        smbios_set_type1_defaults("QEMU", "Standard PC (i440FX + PIIX, 1996)",
                                   args->machine->name);
     }
 
diff --git a/hw/i386/pc_q35.c b/hw/i386/pc_q35.c
index 3ef854a..473f01b 100644
--- a/hw/i386/pc_q35.c
+++ b/hw/i386/pc_q35.c
@@ -102,7 +102,8 @@ static void pc_q35_init(QEMUMachineInitArgs *args)
     }
 
     if (smbios_type1_defaults) {
-        smbios_set_type1_defaults("QEMU", args->machine->desc,
+        /* These values are guest ABI, do not change */
+        smbios_set_type1_defaults("QEMU", "Standard PC (Q35 + ICH9, 2009)",
                                   args->machine->name);
     }
 
-- 
1.7.1

